txnkv/transaction/2pc.go

```diff
@@ -581,8 +606,8 @@ func (c *twoPhaseCommitter) initKeysAndMutations(ctx context.Context) error {
 		if c.txn.schemaAmender != nil || c.txn.assertionLevel == kvrpcpb.AssertionLevel_Off {
 			mustExist, mustNotExist, hasAssertUnknown = false, false, false
 		}
-		c.mutations.Push(op, isPessimistic, mustExist, mustNotExist, it.Handle())
-		size += len(key) + len(value)
+		c.mutations.Push(op, isPessimistic, mustExist, mustNotExist, it.Handle(), cachedValue)
```

Comment:
Some comments are left [here](https://github.com/zyguan/client-go/pull/1/files), maybe avoiding using `nil` checking in the `push` function is safer.

Reply:
`Push` needs to take a parameter of type `[]byte` as the lock->put value, since we do not overwrite membuf anymore (even on `initKeysAndMutations`).